roles/pulp_resource_manager/handlers/main.yml

```diff
@@ -5,3 +5,4 @@
     state: restarted
     daemon_reload: true
   become: true
+  when: pulp_resource_manager_state == 'started'
```

Comment:
I am wondering if this should even uninstall the resource-manager service files.

Reply:
I'm also thinking about it, for me, the ideal is not having it for pulpcore 3.16, but enable it in an easy way when pulpcore < 3.16

Reply:
I see. But also i fear, we are over-engineering this.
Let me ask you two questions:
1. Is it possible to install an older version of pulp (without the devel role) with the installer in version 3.16?
2. Is the only circumstance you are trying to work around here the devel env, where you might need to install anold version of pulpcore?

If you can say "yes" in both cases, then we could just move the RM-service file to the devel-role, and remove any vestige from all other roles.

Reply:
I'm not 100% sure, for pip/released installation I would say the answer is no for both questions.
I believe the source installs are trickier, and I know galaxy uses source installation from git repos, and I don't know if they can use an installer with a previous version of pulpcore.
@Spredzy could help me with these questions?

Reply:
Actually the "needed answers" would have been "no" ,"yes". :facepalm:

Reply:
So the much better approach is to implement this:
https://pulp.plan.io/issues/7006

This way the list of services to restart is defined by the list of services on disk that declare that they are part of the meta "pulpcore" service.

Only the services that are actually installed on the node will be part of it. Which is the brilliant part that makes our lives so much easier in the long run, and enables code cleanup.

It does mean that when pulpcore is upgraded from a prior version, we must immediately stop and remove the service. But once we do, no more hoops to jump through.

Reply:
this feature has been requested by ansible people, so I think it is time to work on it

Reply:
We should still delete the res man service now that we have the meta `pulpcore.service`. Either in a separate PR, or as part of this one.